### PR TITLE
Handle special Celo transactions gracefully

### PR DESCRIPTION
**Pull request by dimriou**
- [x] Add test cases

### CODE REVIEW

File: core/chains/evm/gas/block_history_estimator.go [hunk 1]
```diff
@@ -747,7 +747,7 @@ func (b *BlockHistoryEstimator) getPricesFromBlocks(blocks []evmtypes.Block, eip
 			b.logger.Warnw(fmt.Sprintf("Block %v is not usable, %s", block.Number, err.Error()), "block", block, "err", err)
 		}
 		for _, tx := range block.Transactions {
-			if b.isUsable(tx, b.config, b.logger) {
+			if b.isUsable(tx, block, b.config, b.logger) {
```
Comment (prashantkumar1982) [inline core/chains/evm/gas/block_history_estimator.go:750]: nit: Could we add Info Log when a Tx is deemed unusable? Provide the TxHash in the log message.
Or, perhaps just keep a count of unusable txes here, and for each block, put a log line saying x number of Txs in this block were unusable, out of totak n Txs. Also, atleast log 1 unusable TxHash in that log message, for debugging.

File: core/chains/evm/gas/chain_specific.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 
 // chainSpecificIsUsable allows for additional logic specific to a particular
 // Config that determines whether a transaction should be used for gas estimation
-func chainSpecificIsUsable(tx evmtypes.Transaction, cfg Config) bool {
+func chainSpecificIsUsable(tx evmtypes.Transaction, block evmtypes.Block, cfg Config) bool {
```
Comment (jmank88) [inline core/chains/evm/gas/chain_specific.go:10]: It looks like only `block.BaseFeePerGas` field is used. What about passing that value in directly instead?
```suggestion
func chainSpecificIsUsable(tx evmtypes.Transaction, baseFeePerGas *big.Int, cfg Config) bool {
```

Comment (jmank88) [reply core/chains/evm/gas/chain_specific.go:10]: Why do you say that is more generic with the block type?

The Go idiom here is to accept only what you need to in order to maximize generality (i.e. to be able to call it without constructing an extra block when we don't already have one). We could still make the parameter a generic type either way later on.

File: core/chains/evm/gas/block_history_estimator.go (hunk 1 above)
Comment (prashantkumar1982) [reply core/chains/evm/gas/block_history_estimator.go:750]: Hmm, I just wanted to collect some summary of how many txs in a block were unusable, and perhaps, the reason.
But agree, not asking to do any extra logging or processing.

File: core/chains/evm/gas/chain_specific.go [hunk 3]
```diff
@@ -28,5 +28,19 @@ func chainSpecificIsUsable(tx evmtypes.Transaction, cfg Config) bool {
 			return false
 		}
 	}
+	if cfg.ChainType() == config.ChainCelo {
```
Comment (prashantkumar1982) [inline core/chains/evm/gas/chain_specific.go:31]: How do you ensure that this filtering logic doesn't filter out all the txs in this chain?
What if all txs are either of type 0x7c, or they're all having fees lower than BaseFeePerGas?

File: core/chains/evm/gas/chain_specific.go (hunk 2 above)
Comment (prashantkumar1982) [reply core/chains/evm/gas/chain_specific.go:10]: Ya, makes sense to just use baseFeePerGas for now.
In future if you need other things, can edit it later.

**Review by jmank88 — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]

**Pull request merged**